### PR TITLE
adding `plot_sample_hist` and `create_sample_hist` functions

### LINKED ISSUE #26: Write plot_sample_hist test function

### PR DESCRIPTION
**Pull request by hwilliams10**
Note: a test R script is also included for the `plot_sample_hist` function but it relies on other package functions so it might not work until all functions are included.

### CODE REVIEW

File: R/create_sample_histograms.R [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+#'
+#' @examples
+#' pop <- generate_virtual_pop(100, height, rnorm, 0, 1)
+#' samples <- draw_samples(pop, 3, c(1, 10))
+#' create_sample_histograms(pop, samples, height, c(1, 10))
+create_sample_histograms <- function(pop, samples, var_name, n_s){
+
+  # START INPUT TESTS #
+  # convert var_name to string for testing
+  col_name <- toString(get_expr(enquo(var_name)))
```
Comment (YueJiangMDSV) [inline R/create_sample_histograms.R]: The `get_expr()` function requires `rlang` package. You probably should use rlang::

File: R/create_sample_histograms.R [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    if (!is.element(i, unique(samples$size)))
+      stop("Can only include sample sizes in 'size' column of 'samples' df")
+  }
+
+  # create an empty list of sample histograms
+  samples_hist <- list()
+
+  # for each sample size, generate tidy data needed for histogram plots
+  for (i in 1:length(n_s)){
+    samples_hist[[i]] <- samples %>%
```
Comment (YueJiangMDSV) [inline R/create_sample_histograms.R:56]: Please pay special attention to %>% as Tiffany mentioned during lecture

File: R/create_sample_histograms.R [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      filter(replicate == 1, size == n_s[i]) %>%
```
Comment (YueJiangMDSV) [inline R/create_sample_histograms.R]: The filter function requires dplyr package. You probably should add dplyr::

File: R/create_sample_histograms.R [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      ggplot() +
```
Comment (YueJiangMDSV) [inline R/create_sample_histograms.R]: ggplot should be added in the dependency list. You probably should add ggplot::

**Review by YueJiangMDSV — CHANGES_REQUESTED** [review]

File: tests/testthat/test-create_sample_histograms.R [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# generate helper data
+pop <- generate_virtual_pop(N=100, var_name = height, rexp, 1)
+samples <- draw_samples(pop, 3, c(1, 5, 10))
+x <- create_sample_histograms(pop, samples, height, c(1, 5, 10))
+
+test_that("x axis is mapped to correct variable", {
```
Comment (YueJiangMDSV) [inline tests/testthat/test-create_sample_histograms.R:6]: The tests should be wrapped into one function, instead of a series of different test_that

**Review by YueJiangMDSV — CHANGES_REQUESTED** [review]

File: R/create_sample_histograms.R (hunk 2 above)
Comment (hwilliams10) [reply R/create_sample_histograms.R:56]: Please see latest commit (I imported pipe operator)

File: R/create_sample_histograms.R (hunk 1 above)
Comment (hwilliams10) [reply R/create_sample_histograms.R]: Good suggestion! Incorporated in latest commit

File: R/create_sample_histograms.R (hunk 3 above)
Comment (hwilliams10) [reply R/create_sample_histograms.R]: Good suggestion! Incorporated in latest commit

File: R/create_sample_histograms.R (hunk 4 above)
Comment (hwilliams10) [reply R/create_sample_histograms.R]: Good suggestion! Incorporated in latest commit

File: tests/testthat/test-create_sample_histograms.R (hunk 5 above)
Comment (hwilliams10) [reply tests/testthat/test-create_sample_histograms.R:6]: Are you sure? 
I think it's better to have multiple test thats to give descriptive error messages for each group of tests.

Comment (YueJiangMDSV) [reply tests/testthat/test-create_sample_histograms.R:6]: I understand what you are saying. I just have a vague idea that Tiffany said we should have one function called test_plot_sample_hist, for example, and if you run this function, all the tests in `test_that` would be run. So instead of having a bunch of `test_that`, we can just run one function and test all the things. But now I can't remember where I saw that...

**Review by YueJiangMDSV — CHANGES_REQUESTED** [review]

File: tests/testthat/test-create_sample_histograms.R (hunk 5 above)
Comment (YueJiangMDSV) [reply tests/testthat/test-create_sample_histograms.R:6]: Found it in Milestone2:

"We expect each function to have a unit test function, which is named after the function being tested (e.g., if the function is named foo then the unit test function is named test_foo). The unit test function should test 2 to 3 edge cases to ensure that the function returns what is expected to the user. "

so just wrap everything in one function. Then I think everything is fine.

File: tests/testthat/test-plot_sample_hist.R [hunk 6]
```diff
@@ -0,0 +1,19 @@
+test_that("get expected number of grobs", {
```
Comment (YueJiangMDSV) [inline tests/testthat/test-plot_sample_hist.R:1]: Same as above. Just wrap everything in one function. Then I think everything is fine.

**Review by YueJiangMDSV — CHANGES_REQUESTED** [review]

**Review by YueJiangMDSV — APPROVED** [review]
